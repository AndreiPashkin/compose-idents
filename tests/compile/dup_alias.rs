use compose_idents::compose_idents;

compose_idents!(
    my_fn = concat(foo, _, "baz"),
    my_fn = concat(gork, _, bork),
    {
        fn my_fn() -> u32 {
            42
        }
    }
);

fn main() {
    assert_eq!(foo_baz(), 42);
}
