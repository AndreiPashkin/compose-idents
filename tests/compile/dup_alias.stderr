error: Alias "my_fn" is already defined
 --> tests/compile/dup_alias.rs:3:65
  |
3 | compose_idents!(my_fn = [foo, _, "baz"], my_fn = [gork, _, bork], {
  |                                                                 ^

error[E0425]: cannot find function `foo_baz` in this scope
  --> tests/compile/dup_alias.rs:10:16
   |
10 |     assert_eq!(foo_baz(), 42);
   |                ^^^^^^^ not found in this scope
